cmd/bucket-replication.go

```diff
@@ -1861,16 +1865,34 @@ func (p *ReplicationPool) ResizeFailedWorkers(n int) {
 	}
 }
 
+const (
+	minLargeObjSize = 128 * humanize.MiByte // 128MiB
+	maxLargeWorkers = 10
+)
+
 // getWorkerCh gets a worker channel deterministically based on bucket and object names.
 // Must be able to grab read lock from p.
-func (p *ReplicationPool) getWorkerCh(bucket, object string) chan<- ReplicationWorkerOperation {
+// De-prioritize larger uploads by capping the workers that can handle them.
+
+func (p *ReplicationPool) getWorkerCh(bucket, object string, sz int64) chan<- ReplicationWorkerOperation {
 	h := xxh3.HashString(bucket + object)
 	p.mu.RLock()
 	defer p.mu.RUnlock()
-	if len(p.workers) == 0 {
+	workerCnt := uint64(len(p.workers))
+	if workerCnt == 0 {
 		return nil
 	}
-	return p.workers[h%uint64(len(p.workers))]
+	largeWorkerCount := uint64(math.Min(maxLargeWorkers, float64(sz/minLargeObjSize)))
```

Comment:
This is soo complicated. My head hurts ;)

Can't we just allocate 10 workers separately and use these for big objects exclusively?

Reply:
trying to avoid creating yet more workers and managing their auto scaling separately

Reply:
@poornas Can't we just create 10 static workers?

Reply:
we could. changed the PR